src/main/utils/shell-env.ts

```diff
@@ -64,3 +64,13 @@ export async function shellEnv(shell?: string, forceRetry = false) : Promise<Env
 
   return envVars;
 }
+
+export function resolveEnv(value: string, env: Record<string, any>): string {
+  // resolves both $VAR and ${VAR} syntax, and makes sure to
+  // ignore malformed syntax, such as begins with digit or lower case characters.
+  return value.replace(/\$\{?([A-Z_][A-Z0-9_]*)(?:(?<=\$\{\1)\})?/g, function (match) {
```

Comment:
It is incorrect that lowercase values are invalid in the environment.

Furthermore shouldn't this regex has a `^` at the end? Lastly this regex is incorrect as it matches `${A`. Furthermore there is no need for it to be a global regex.

To continue bracket matching is not a regular language and requires at least a context free grammer to properly match.

Reply:
I think that we should just support `$...` and not `${..}` and this information should be added to the hint of the "Working Directory" setting.

Reply:
Agreed, will update, thanks.

Reply:
I pushed some changes to now only match on simple syntax (`$HOME`, `$var`), and I added some tests to validate the logic.

Also the reason for a global regex and no line start anchor (`^`) is I thought it would be a good idea while at it to support more than one var, although I have no problem restricting it.

Finally, I updated the hint of "Working Directory" setting to mention environment variable expansion.

Reply:
Will rereview. If we wanted to do some sort of hierarchy then I think we would switch to an array. No need to do weird string parsing.

Reply:
Yeah the more I think of it, the less I think its useful. The idea was just to allow arbitrary vars expansion, like: `$HOME/myfolder/$ENV` to resolve to `/home/alice/myfolder/development`.

Let me know if you just want me to update this to only validate that cwd exists instead. Thanks.

Reply:
After discussing it with some other team members we think that the expansion of vars is a very good idea but probably should be its own PR because it is is a feature enhancement whereas the rest of this PR is a bug fix.

Reply:
Awesome I will do the split and keep only the bug fix here.

I'll open another PR to add var expansion, and we can discuss there the best way to implement, thanks :)